src/js/workshop-comments.js

```diff
@@ -0,0 +1,138 @@
+
+	// Hide required and comment submission message.
+	// Note: It's necessary to hide "required" and "succesful/failed" submitted comment message indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment.
+	document.querySelectorAll(".required-name, .required-comment, .submit-success-message, .submit-failure-message").forEach(element => {
+		applyVisibility(element, false);
+	});
+
+	// Verify input Fields.
+	let isValueMissing = false;
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		isValueMissing = true;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		isValueMissing = true;
+	}
+
+	if (isValueMissing) {
+		return;
+	}
+
+	// Disable input fields when proccessing a new request.
+	disableFields(true);
+
+	const XHR = new XMLHttpRequest();
+
+	// Bind the FormData object and the form element.
+	const commentFormData = new FormData( commentForm );
+
+	// Submit and add comment to top of comment section of page on successful data submission.
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			showSubmittedComment(name, comment);
+			// Show successful comment submission message.
+			applyVisibility(document.querySelector(".submit-success-message"), true , "block");
+			// Clear Text Fields.
+			commentForm.reset();
+		} else {
+			// Show failed comment submission message.
+			submitFailureMessage.setAttribute("style", "display: block;");
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error.
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message.
+		submitFailureMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request.
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form.
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
+	XHR.setRequestHeader("Content-Type", "application/json");
+	XHR.send(jsonFormData);
+});
+
+/*
+ * Hide/show required and success messages.
+ * @param {DOMElement} domSelector - The DOM Element to parse.
+ * @param {Boolean} toShow - value equals true to show and equals false to hide.
+ * @param {String}[Optional] displayType - CSS display value to apply. Only applicable when toShow is set to true.
+ */
+function applyVisibility(domSelector, toShow, displayType) {
+	displayType = toShow ? displayType : "none";
+	domSelector.setAttribute("style", `display:  ${displayType};`);
+}
+
+/*
+ * Display the new comment on the current page.
+ * @param {DOMElement} nameField - DOM element of name field in comment form.
+ * @param {DOMElement} commentField - DOM element of comment field in comment form.
+ */
+function showSubmittedComment(nameField, commentField) {
```

Comment:
It's probably better to change "nameField" to "name" and "commentField" to "comment" as they are values instead of fields.